city_scrapers/spiders/il_sex_offender_management.py

```diff
@@ -0,0 +1,234 @@
+import json
+import re
+from datetime import datetime, timedelta
+from io import BytesIO, StringIO
+
+import scrapy
+from city_scrapers_core.constants import BOARD
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from pdfminer.high_level import extract_text_to_fp
+from pdfminer.layout import LAParams
+
+
+class IlSexOffenderManagementSpider(CityScrapersSpider):
+    name = "il_sex_offender_management"
+    agency = "Illinois Sex Offender Management Board"
+    timezone = "America/Chicago"
+    custom_settings = {"ROBOTSTXT_OBEY": False}
+    start_urls = [
+        "https://www2.illinois.gov/idoc/Pages/SexOffenderManagementBoard.aspx"
+    ]  # noqa
+    meeting_minutes = []
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+        # first go through and gather list of lists for meeting minutes
+        for link in response.css(".soi-article-content a"):
+            link_text = " ".join(link.css("*::text").extract())
+            if ("meeting minutes") in link_text.lower():
+                # create new list holding meeting minutes date and link
+                date_time_obj = self._get_datetime_obj_meeting_minutes(
+                    link_text
+                )  # noqa
+                href = response.urljoin(link.attrib["href"])
+                self.meeting_minutes.append([date_time_obj, href])
+
+        for link in response.css(".soi-article-content a"):
+            link_text = " ".join(link.css("*::text").extract())
+            if "meeting" not in link_text.lower():
+                continue
+            if "agenda" in link_text.lower():
+                # scrape agenda pdf for meeting information
+                yield scrapy.Request(
+                    response.urljoin(link.attrib["href"]),
+                    callback=self._parse_documents,
+                    dont_filter=True,
+                )
+            # special case where link to video was uploaded rather than agenda
+            if "webex" in link_text.lower():
+                yield scrapy.Request(
+                    link.attrib["href"], callback=self._parse_meta_video
+                )
+
+    def _parse_meta_video(self, response):
+        """Parse through meta of response in JSON if link provided rather than PDF"""
+        meta_text = response.css(
+            "[type='application/json']#extendedData::text"
+        ).extract_first()
+        if not meta_text:
+            return
+        meta_text = meta_text.strip()
+        clean_text_vid = json.loads(meta_text)
+
+        start_time, end_time = self._video_start_end_time(clean_text_vid)
+
+        meeting = Meeting(
+            title="MANAGEMENT BOARD",
```

Comment:
We want these to be formatted for display, so we can do "Management Board" instead of all-caps here